ui/components/ui/custom/custom-dropdown-filter.tsx

```diff
@@ -192,18 +194,37 @@ export const CustomDropdownFilter: React.FC<CustomDropdownFilterProps> = ({
                 hideScrollBar
                 className="flex max-h-96 max-w-56 flex-col gap-y-2 py-2"
               >
-                {memoizedFilterValues.map((value) => (
-                  <Checkbox
-                    classNames={{
-                      label: "text-small font-normal",
-                      wrapper: "checkbox-update",
-                    }}
-                    key={value}
-                    value={value}
-                  >
-                    {value}
-                  </Checkbox>
-                ))}
+                {memoizedFilterValues.map((value) => {
+                  // Find the corresponding entity from valueLabelMapping
+                  const matchingEntry = filter.valueLabelMapping?.find(
+                    (entry) => entry[value],
+                  );
+                  const entity = matchingEntry?.[value];
+
+                  return (
+                    <Checkbox
+                      classNames={{
+                        label: "text-small font-normal",
+                        wrapper: "checkbox-update",
+                      }}
+                      key={value}
+                      value={value}
+                    >
+                      {entity ? (
+                        <div>
```

Comment:
Is this `div` necessary ? let's avoid it if we can.

Edit: Just came to mind, can we use this div to add a hover effect? Right now, when hovering over the value (EntityInfoShort), there's no effect. Could we add a smooth hover effect? 

Thanks!